<launch>
  <param name="use_sim_time" value="false" />
  <arg name="map_id" />
  <arg name="lidar_type" />

  <group if="$(eval arg('lidar_type') == 'velodyne')">
    <node pkg="ndt_localization" type="ndt_localization_node" name="ndt_localization_node" output="screen">
      <rosparam file="$(find ndt_localization)/config/ndt.yaml" command="load" />
      <param name="global_map_file" value="$(find ndt_localization)/map/$(arg map_id).pcd"/>
      <param name="laser_frame" value="velodyne" />
      <param name="lidar_topic" value="velodyne_points" />
    </node>
  </group>

  <group if="$(eval arg('lidar_type') == 'rslidar')">
    <node pkg="ndt_localization" type="ndt_localization_node" name="ndt_localization_node" output="screen">
      <rosparam file="$(find ndt_localization)/config/ndt.yaml" command="load" />
      <param name="global_map_file" value="$(find ndt_localization)/map/$(arg map_id).pcd"/>
      <param name="laser_frame" value="rslidar" />
      <param name="lidar_topic" value="rslidar_points" />
    </node>
  </group>

  <node name="MapServer" pkg="map_server" type="map_server" args="$(find ndt_localization)/map/$(arg map_id).yaml"/>
</launch>
